### PR TITLE
test: Add HACKING.md on how to run end to end test locally

### PR DESCRIPTION
**Pull request by henrywang**

### CODE REVIEW

File: test/end-to-end/HACKING.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+## Getting the development dependencies
+
+Cockpit-composer end to end test uses Node.js during case development.
+To make changes on Cockpit you'll want to install Node.js, NPM and
+various development dependencies.
+
+On Fedora:
+
+    $ sudo dnf install nodejs
```
Comment (martinpitt) [inline test/end-to-end/HACKING.md]: Shouldn't that be "npm"? That pulls in nodejs, but nodejs itself doesn't have /bin/npm.

File: test/end-to-end/HACKING.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+And lastly get development dependencies:
+
+    $ npm install
+
+## Preparing composer for testing on local machine
+
+It's easy to set up your local Linux machine for rapid deployment composer.
+First install `cockpit` and `lorax-composer` on your local machine:
+
+On Fefora:
```
Comment (martinpitt) [inline test/end-to-end/HACKING.md]: Fe"d"ora

File: test/end-to-end/HACKING.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    $ sudo dnf install nodejs
+
+And lastly get development dependencies:
+
+    $ npm install
+
+## Preparing composer for testing on local machine
+
+It's easy to set up your local Linux machine for rapid deployment composer.
```
Comment (martinpitt) [inline test/end-to-end/HACKING.md]: Not sure what "rapid deployment composer" means?

File: test/end-to-end/HACKING.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+On Fefora:
+
+    $ sudo dnf install cockpit lorax-composer
+
+Next run this command from your top level Cockpit-composer checkout
+directory, and make sure to run it as the same user that you'll
+use to log into Cockpit below.
+
+    $ mkdir -p ~/.local/share/
+    $ ln -s ../../dist ~/.local/share/welder
```
Comment (martinpitt) [inline test/end-to-end/HACKING.md]: This is done by `make devel-install`.

Also, this just repeats what's already in README.md and this isn't really related to the end-to-end tests specifially, so IMHO this whole paragraph should stay in README.md.

**Comment by henrywang** [comment]
Thanks @martinpitt!!!! Updated!

File: test/end-to-end/HACKING.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    $ npm install
+
+Lastly get `cockpit`, `lorax-composer`, and `cockpit-composer` running
+locally.
+
+## Setup selenium and browser running environment
+
+End to end test depends on [webdriver.io](http://v4.webdriver.io/), which
+is a WebDriver test framework for Node.js. A selenium running environment
+should be ready before test running. The easist and fastest way to setup
```
Comment (KKoukiou) [inline test/end-to-end/HACKING.md]: s/easist/easiest/

Comment (henrywang) [reply test/end-to-end/HACKING.md]: Thanks @KKoukiou for catching this.

**Pull request merged**